packages/core/src/network.js

```diff
@@ -6,70 +6,87 @@ import {
   createRequestFailedHandler
 } from './discovery';
 
-const NETWORK_TIMEOUT = 30000;
-
 // The Interceptor class creates common handlers for dealing with intercepting asset requests
 // for a given page using various devtools protocol events and commands.
 export default class Network {
+  static TIMEOUT = 30000;
+
+  log = logger('core:network');
+
   #pending = new Map();
   #requests = new Map();
   #intercepts = new Map();
   #authentications = new Set();
-  #frames = new Map();
 
-  log = logger('core:network');
-
-  constructor(page) {
+  constructor(page, options) {
     this.page = page;
-    this.page.on('Fetch.authRequired', this._handleAuthRequired);
-    this.page.on('Fetch.requestPaused', this._handleRequestPaused);
-    this.page.on('Network.requestWillBeSent', this._handleRequestWillBeSent);
-    this.page.on('Network.responseReceived', this._handleResponseReceived);
-    this.page.on('Network.eventSourceMessageReceived', this._handleEventSourceMessageReceived);
-    this.page.on('Network.loadingFinished', this._handleLoadingFinished);
-    this.page.on('Network.loadingFailed', this._handleLoadingFailed);
-    this.page.on('Page.frameDetached', this._handleFrameDetached);
-    this.page._handleCloseRace(this.page.send('Network.enable'));
+    this.timeout = options.networkIdleTimeout ?? 100;
```

Comment:
Does this differ from our config defaults?

Reply:
No, it's the same. However, our config default will always take precedence over this default. This default is used when borrowing a browser page in other SDKs that don't provide their own idle timeout.